java/core/src/main/java/sleeper/core/schema/Schema.java

```diff
@@ -185,7 +184,12 @@ public Builder valueFields(Field... valueFields) {
         }
 
         public Schema build() {
-            return new Schema(this);
+            validateNoNullFields(rowKeyFields, sortKeyFields, valueFields);
```

Comment:
This call, and the initialisation of the fields, seems to conflict with the intended behaviour in validateRowKeys, validateSortKeys and validateValueKeys. Those methods remove nulls explicitly and replace them with empty lists.

We can pick one approach, to stop it looking like it's happening one way when it's actually not. We could just set rowKeyFields, sortKeyFields and valueFields to the result of the validate methods here instead.